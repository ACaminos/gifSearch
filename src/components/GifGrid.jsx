import React from 'react'

export const GifGrid = ( { category } ) => {

  const getGifs = async() => {
    const url = `https://api.giphy.com/v1/gifs/search?api_key=UKDssLvF2UM3gWSFMDMI8gsEdXRX0cpP&q=${ category }`
    const resp = await fetch( url )

    console.log( resp )
  }
  return (
    <>
        <h3>{ category }</h3>
        <p>Que onda</p>
    </>
  )
}
