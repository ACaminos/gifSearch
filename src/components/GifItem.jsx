import propTypes from "prop-types"

export const GifItem = ( { title, url, id } ) => {
  return (
    <div className="card">
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
  )
}

GifItem.propTypes = {
  url : propTypes.string.isRequired,
  title : propTypes.string.isRequired,
  id : propTypes.string
}