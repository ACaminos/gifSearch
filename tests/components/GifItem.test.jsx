import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('test GifItem', () => {
    const url = 'http://localhost/www.hola.com.ar';
    const title = 'Hola como va';

    test('Debe dar ok en recibir url y title y hacer snapshot', () => {
        const { container } = render( <GifItem url={ url } title={ title } /> );
        expect ( container ).toMatchSnapshot();
    })

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render( <GifItem url={ url } title={ title } /> );
        // screen.debug()
        // expect( screen.getByRole('img').src ).toBe( url )
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    })

    test('Debe de mostrar el titulo en el componente', () => {
        render( <GifItem url={ url } title={ title } /> );
        expect( screen.getByText( title ) ).toBeTruthy();
    })
});
