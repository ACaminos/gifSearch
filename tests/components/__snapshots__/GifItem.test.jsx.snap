// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GifItem Debe dar ok en recibir url y title y hacer snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola como va"
      src="http://localhost/www.hola.com.ar"
    />
    <p>
      Hola como va
    </p>
  </div>
</div>
`;
